source/databricks/tests/test_time_series_transformer.py

```diff
@@ -0,0 +1,138 @@
+                    "Points": [
+                        {
+                            "Position": 1,
+                            "Quality": 3,
+                            "Quantity": Decimal(1.1),
+                        },
+                        {
+                            "Position": 2,
+                            "Quality": 3,
+                            "Quantity": Decimal(1.1),
+                        },
+                    ],
+                    "Resolution": Resolution,
+                    "StartDateTime": StartDateTime,
+                },
+                "Product": "1",
+                "Receiver": {
+                    "BusinessProcessRole": 0,
+                    "Id": "2",
+                },
+                "RegistrationDateTime": RegistrationDateTime,
+                "Sender": {
+                    "BusinessProcessRole": 0,
+                    "Id": "1",
+                },
+                "SeriesId": "1",
+                "TransactionId": "1",
+                "year": 2022,
+                "month": 6,
+                "day": 9,
+            }
+        ]
+
+        return spark.createDataFrame(df, time_series_unprocessed_schema)
+
+    return factory
+
+
+@pytest.mark.parametrize(
+    "registration_date_time, expected_registration_date_time",
+    [
+        (None, "2022-06-09T12:09:15.000Z"),
+        ("2022-06-10T12:09:15.000Z", "2022-06-10T12:09:15.000Z"),
+    ],
+)
+def test__transform_unprocessed_time_series_to_points__registration_date_time_fallsback_to_created_date_time_when_none(
+    time_series_unprocessed_factory,
+    timestamp,
+    registration_date_time,
+    expected_registration_date_time,
+):
+    # Arrange
+    time_series_unprocessed_df = time_series_unprocessed_factory(
+        RegistrationDateTime=timestamp(registration_date_time),
+        CreatedDateTime=timestamp("2022-06-09T12:09:15.000Z"),
+    )
+
+    # Act
+    actual_df = transform_unprocessed_time_series_to_points(time_series_unprocessed_df)
+    actual_registration_data_time = actual_df.collect()[0]["RegistrationDateTime"]
+
+    # Assert
+    assert actual_registration_data_time == timestamp(expected_registration_date_time)
+
+
+@pytest.mark.parametrize(
+    "resolution, expected_time_for_position_2",
+    [
+        (1, "2022-06-08T12:15:00.000Z"),
+        (2, "2022-06-08T13:00:00.000Z"),
+        (3, "2022-06-09T12:00:00.000Z"),
+        (4, "2022-07-08T12:00:00.000Z"),
+    ],
+)
+def test__transform_unprocessed_time_series_to_points__sets_correct_time_depending_on_resolution(
+    time_series_unprocessed_factory, timestamp, resolution, expected_time_for_position_2
+):
+    # Arrange
+    time_series_unprocessed_df = time_series_unprocessed_factory(
+        StartDateTime=timestamp("2022-06-08T12:00:00.000Z"), Resolution=resolution
+    )
+
+    # Act
+    actual_df = transform_unprocessed_time_series_to_points(time_series_unprocessed_df)
+    actual_time_for_position_2 = actual_df.collect()[1]["time"]
```

Comment:
Is this the same as?

```suggestion
    actual_time_for_position_2 = actual_df.first().time
```

Reply:
.first() will take the row at position 0, so the first row and collect()[1] will take the row at position 1, so the second row